### ISSUE TITLE
`import-url`/`import`: follow ups for cloud versioning

### ISSUE THREAD

**Issue by dberenbaum** [iterative/dvc#8464]
- [ ] For `dvc update` we can probably get around downloading when it hasn't changed, but that's a separate feature request I think (https://github.com/iterative/dvc/issues/8249#issuecomment-1288500204_)
~- [ ] For chained `import-url` imports I think it will not work right now unless the file has been pushed to a DVC remote, but when we get to refactoring regular cached objects to use dvc-data index for push/pull (so everything is unified to use the dvc-data index) this kind of chaining (from original source location) should work out of the box. (https://github.com/iterative/dvc/issues/8249#issuecomment-1288500204_)~ Moved to #8789
~- [ ] Add support to import/get from another repo where that repo's remote is cloud-versioned.~ Moved to #8789
- [x] Don't push `dvc import` data to remote when remote is configured with worktree = true. (https://github.com/iterative/dvc/pull/8581)
- [x] With cloud versioning enabled, DVC always returns that it has pushed all files, even for files that aren't pushed (including files that have already been pushed and files with `push: false`). (https://github.com/iterative/dvc/pull/8606)

**Comment by dberenbaum** [comment]
> * For `dvc update` we can probably get around downloading when it hasn't changed

@pmrowla Thoughts on this? This would be nice to have for initial release, since redownloading on every update seems unrealistic in any real scenario.

**Comment by dberenbaum** [comment]
@karajan1001 @pmrowla There was a regression that was making update download unchanged datasets even for regular remotes that was fixed in https://github.com/iterative/dvc/pull/8752. For cloud-versioned remotes, update still downloads unchanged datasets even after that fix.

**Comment by dberenbaum** [comment]
We should also check that data added to a worktree remote doesn't redownload on `dvc update` if nothing changed.

**Comment by dberenbaum** [comment]
For `update` after `import-url` without cloud versioning, here's what I see:

https://user-images.githubusercontent.com/2308172/213193601-cf076be7-898f-4f36-8b08-0cadac0eb110.mov

For `update` after `import-url --version-aware`, I see that everything gets downloaded again:

https://user-images.githubusercontent.com/2308172/213193702-224b845a-d2a7-4d77-8bca-c1e735ed1f01.mov

For a non-imported dataset pushed to a `worktree` remote, it looks like nothing gets downloaded, but it would be nice to have this be more explicit and match the UI of the first video:

https://user-images.githubusercontent.com/2308172/213193947-8b58d397-d6b5-4dbb-bd2a-b5ce499d1c7c.mov

**Comment by dberenbaum** [comment]
@karajan1001 Any updates on this?

**Comment by karajan1001** [comment]
> @karajan1001 Any updates on this?

I'm sorry, I'm still debugging what's wrong with my s3 setup.

**Comment by karajan1001** [comment]
@dberenbaum A new situation. I found that the problem of duplicated downloading had already been solved in #8849, but after #8882 it became terribly slow (takes several minutes to finish).

**Comment by daavoo** [comment]
> but after https://github.com/iterative/dvc/pull/8882 it became terribly slow (takes several minutes to finish).

@karajan1001  not sure what is `it` in the phrase. `dvc update` after `dvc import-url --version-aware` ?  

Could you share the profile from:

```
pip install viztracer
dvc update --viztracer --viztracer-depth 12
```

**Comment by karajan1001** [comment]
> @karajan1001 not sure what is it in the phrase. dvc update after dvc import-url --version-aware ?

Yes, as well as `import` itself, but maybe it is caused by the network problem on my side,

**Comment by dberenbaum** [comment]
@karajan1001 I'm not seeing a major performance difference, but could you and @daavoo debug and make sure it's not a problem? We can close the issue once that's addressed.

**Comment by karajan1001** [comment]
A big difference is that in the new version of `dvc-data`'s get method, we have `find` in it.

![image](https://user-images.githubusercontent.com/6745454/217225279-41ea5983-3313-43c2-849f-170d88bcc1d9.png)
![image](https://user-images.githubusercontent.com/6745454/217225314-1a8c4231-cf36-4f91-9517-12cc93b60caf.png)

I found today, the downloading problem appears again even in the latest version, looks like #8849 only partially solved it. Need a more detailed investigation.

**Comment by pmrowla** [comment]
The `update` after `import-url` downloading files when nothing has changed issue should be fixed in main (underlying cause was also related to the `import --no-download` bug)

**Comment by karajan1001** [comment]
I can confirm that the redownloading problem been solved.

**Comment by dberenbaum** [comment]
@karajan1001 Are there still performance issues to investigate? Otherwise, let's close this issue.

**Comment by karajan1001** [comment]
> @karajan1001 Are there still performance issues to investigate? Otherwise, let's close this issue.

The performance down was because, in the previous version, we didn't consider the s3 fs as a version_aware fs, and after #8882 this bug was fixed, but we still have performance problems on a `version_aware`  S3FileSystem, But as it only affects users under particular network conditions. I recommend that we close this one and open a new `p2` one for the performance problem.
